Sources/AsyncHTTPClient/HTTPClient.swift

```diff
@@ -556,7 +556,10 @@ public class HTTPClient {
             }
         }.always { _ in
             setupComplete.succeed(())
-        }.cascadeFailure(to: task.promise)
+        }.whenFailure { error in
```

Comment:
Are we confident that this event loop is an acceptable one to use to call into the delegate? We aren't paying any attention to the preference here. Seems like we need a `.hop(to: taskEL)` here.

Reply:
Seconded. This also needs a test case. Also, I think there's a special method to call out to delegates that takes care of the hopping so we don't get it wrong. Essentially, all naked calls to the delegate should really go away and use the methods that are meant for that.

Reply:
Test case for correct EL?

Reply:
Make a test case that would fail _without_ the necessary hopping.

Reply:
done